core/embed/trezorhal/optiga/optiga.c

```diff
@@ -89,3 +90,22 @@ bool optiga_read_cert(uint8_t index, uint8_t *cert, size_t max_cert_size,
                                              cert, max_cert_size, cert_size);
   return OPTIGA_SUCCESS == ret;
 }
+
+bool optiga_random_buffer(uint8_t *dest, size_t size) {
+  while (size > OPTIGA_RANDOM_MAX_SIZE) {
+    if (optiga_get_random(dest, OPTIGA_RANDOM_MAX_SIZE) != OPTIGA_SUCCESS) {
+      return false;
+    }
+    dest += OPTIGA_RANDOM_MAX_SIZE;
+    size -= OPTIGA_RANDOM_MAX_SIZE;
+  }
+
+  if (size < OPTIGA_RANDOM_MIN_SIZE) {
+    static uint8_t buffer[OPTIGA_RANDOM_MIN_SIZE] = {0};
+    optiga_result ret = optiga_get_random(buffer, OPTIGA_RANDOM_MIN_SIZE);
+    memcpy(dest, buffer, size);
+    return ret == OPTIGA_SUCCESS;
+  }
```

Comment:
Rereading the code, I realized these two blocks need to be swapped, otherwise requesting for example between 257-263 bytes fails. Done in 8f20a0177f86eadb2cab64ba9cc7045ce0dbbca0.